### PR TITLE
doc/website: add serial types to knowledge pages

### PR DESCRIPTION
**Pull request by a8m**

### CODE REVIEW

File: doc/md/knowledge/postgres/serial-columns.md [hunk 1]
```diff
@@ -0,0 +1,258 @@
+---
+id: serial-columns
+title: Serial Type Columns in PostgreSQL
+slug: /knowledge/postgres/serial-columns
+---
+
+PostgreSQL allows creating columns of types `smallserial`, `serial`, and `bigserial`. However, these types are not
```
Comment (hilakashai) [inline doc/md/knowledge/postgres/serial-columns.md:7]:
```suggestion
PostgreSQL allows creating columns of types `smallserial`, `serial`, and `bigserial`. These types are not
```

File: doc/md/knowledge/postgres/serial-columns.md [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+actual types but more like "macros" for creating non-nullable integer columns with sequences attached.
```
Comment (hilakashai) [inline doc/md/knowledge/postgres/serial-columns.md:8]:
```suggestion
_actual_ types, but more like "macros" for creating non-nullable integer columns with sequences attached.
```

File: doc/md/knowledge/postgres/serial-columns.md [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+
+We can see it in action by creating a table with 3 "serial columns":
```
Comment (hilakashai) [inline doc/md/knowledge/postgres/serial-columns.md:10]:
```suggestion
We can see this in action by creating a table with 3 "serial columns": 
```

File: doc/md/knowledge/postgres/serial-columns.md [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+
+```sql title="Serials Description"
+ Column |   Type   | Nullable |            Default            
+--------+----------+----------+-------------------------------
+ c1     | smallint | not null | nextval('t_c1_seq'::regclass)
+ c2     | integer  | not null | nextval('t_c2_seq'::regclass)
+ c3     | bigint   | not null | nextval('t_c3_seq'::regclass)
+```
+
+As you can see, each serial column was created as non-nullable integer with a default value set to next sequence
```
Comment (hilakashai) [inline doc/md/knowledge/postgres/serial-columns.md:28]:
```suggestion
As you can see, each serial column was created as non-nullable integer with a default value set to the next sequence
```

File: doc/md/knowledge/postgres/serial-columns.md [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+value.
+
+:::info
+Note that `nextval` increments the sequence by 1 and returns its value. Thus, the first call to
+`nextval('serials_c1_seq')` returns 1, the second returns 2, etc.
+:::
+
+### `ALTER COLUMN` type to serial
+
+Sometimes, it is necessary to change the column type from `integer` type to `serial`. However, as mentioned above, the
```
Comment (hilakashai) [inline doc/md/knowledge/postgres/serial-columns.md:38]:
```suggestion
Sometimes it is necessary to change the column type from `integer` type to `serial`. However, as mentioned above, the
```

File: doc/md/knowledge/postgres/serial-columns.md [hunk 6]
```diff
@@ -0,0 +1,258 @@
...
+// highlight-next-line-error-message
+ERROR: type "serial" does not exist
+```
+
+We can achieve this by manually creating a [sequence](https://www.postgresql.org/docs/current/sql-createsequence.html)
+owned by the column `c`, and setting the column `DEFAULT` value to the incremental counter of the sequence using the
+[`nextval`](https://www.postgresql.org/docs/current/functions-sequence.html) function.
+
+:::note
+Note that, it is recommended to follow the PostgreSQL naming format (i.e. `<table>_<column>_seq`)
```
Comment (hilakashai) [inline doc/md/knowledge/postgres/serial-columns.md:56]:
```suggestion
Note that it is recommended to follow the PostgreSQL naming format (i.e. `<table>_<column>_seq`)
```

File: doc/md/knowledge/postgres/serial-columns.md [hunk 7]
```diff
@@ -0,0 +1,258 @@
...
+when creating the sequence as some database tooling know to detect such columns as "serial columns".
```
Comment (hilakashai) [inline doc/md/knowledge/postgres/serial-columns.md:57]:
```suggestion
when creating the sequence as some database tools know to detect such columns as "serial columns".
```

File: doc/md/knowledge/postgres/serial-columns.md [hunk 8]
```diff
@@ -0,0 +1,258 @@
...
+-- Fails!
+INSERT INTO "t" DEFAULT VALUES;
+// highlight-next-line-error-message
+ERROR:  duplicate key value violates unique constraint "t_pkey"
+// highlight-next-line-error-message
+DETAIL:  Key (c)=(2) already exists.
+```
+
+We can work around this by setting the sequence current value to the maximum value of `c`, so the following call to
+`nextval` will return `MAX(c)+1`, the one after `MAX(c)+2`, etc.
```
Comment (hilakashai) [inline doc/md/knowledge/postgres/serial-columns.md:94]:
```suggestion
`nextval` will return `MAX(c)+1`, the one after `MAX(c)+2`, and so on.
```

File: doc/md/knowledge/postgres/serial-columns.md [hunk 9]
```diff
@@ -0,0 +1,258 @@
...
+ 2
+ 3
+ 4
+// highlight-end
+```
+
+### Managing Serial Columns with Atlas
+
+Atlas makes it easier to define and manipulate columns of `serial` types. Let's use the
+[`atlas schema inspect`](../../cli/reference.md#atlas-schema-inspect) command to get a representation in Atlas
```
Comment (hilakashai) [inline doc/md/knowledge/postgres/serial-columns.md:119]:
```suggestion
[`atlas schema inspect`](../../cli/reference.md#atlas-schema-inspect) command to get a representation in 
```

File: doc/md/knowledge/postgres/serial-columns.md [hunk 10]
```diff
@@ -0,0 +1,258 @@
...
+ 3
+ 4
+// highlight-end
+```
+
+### Managing Serial Columns with Atlas
+
+Atlas makes it easier to define and manipulate columns of `serial` types. Let's use the
+[`atlas schema inspect`](../../cli/reference.md#atlas-schema-inspect) command to get a representation in Atlas 
+the Atlas language of the table we created above:
```
Comment (hilakashai) [inline doc/md/knowledge/postgres/serial-columns.md:120]: 'the Atlas language' sounds kind of weird, I'd write in the Atlas HCL format or something

File: doc/md/knowledge/postgres/serial-columns.md [hunk 11]
```diff
@@ -0,0 +1,258 @@
...
+  }
+  primary_key {
+    columns = [column.c]
+  }
+}
+schema "public" {
+}
+```
+
+After inspecting the schema, we can modify it a bit to demonstrate Atlas's capabilities in migration planning:
```
Comment (hilakashai) [inline doc/md/knowledge/postgres/serial-columns.md:141]:
```suggestion
After inspecting the schema, we can modify it to demonstrate Atlas's capabilities in migration planning:
```

File: doc/md/knowledge/postgres/serial-columns.md [hunk 12]
```diff
@@ -0,0 +1,258 @@
...
+  }
+  primary_key {
+    columns = [column.c]
+  }
+}
+schema "public" {
+}
+```
+
+After changing the `c` to `bigint`, we can run `schema apply` and let Atlas plan and execute the new changes:
```
Comment (hilakashai) [inline doc/md/knowledge/postgres/serial-columns.md:196]:
```suggestion
After changing column `c` to `bigint`, we can run `schema apply` and let Atlas plan and execute the new changes:
```

File: doc/md/knowledge/postgres/serial-columns.md [hunk 13]
```diff
@@ -0,0 +1,258 @@
...
+-- Modify "t" table
+// highlight-next-line-info
+ALTER TABLE "public"."t" ALTER COLUMN "c" DROP DEFAULT
+-- Drop sequence used by serial column "c"
+// highlight-next-line-info
+DROP SEQUENCE IF EXISTS "public"."t_c_seq"
+✔ Apply
+```
+
+As you can see, Atlas dropped the `DEFAULT` value that was created by the `serial` type, but in addition it removed
```
Comment (hilakashai) [inline doc/md/knowledge/postgres/serial-columns.md:211]:
```suggestion
As you can see, Atlas dropped the `DEFAULT` value that was created by the `serial` type, and in addition removed
```

File: doc/md/knowledge/postgres/serial-columns.md [hunk 14]
```diff
@@ -0,0 +1,258 @@
...
+also the sequence that was attached to it as it is no longer used by the column.
```
Comment (hilakashai) [inline doc/md/knowledge/postgres/serial-columns.md:212]:
```suggestion
the sequence that was attached to it, as it is no longer used by the column.
```

File: doc/md/knowledge/postgres/serial-columns.md [hunk 15]
```diff
@@ -0,0 +1,258 @@
...
+  }
+}
+schema "public" {
+}
+```
+
+Changing a column type from `bigint` to `serial` requires 3 changes:
+1. Create a sequence named `t_c_seq` owned by `c`.
+2. Set the `DEFAULT` value of `c` to `nextval('"public"."t_c_seq"')`.
+3. Alter the column type as `serial` maps to `integer` (!= `bigint`).
```
Comment (hilakashai) [inline doc/md/knowledge/postgres/serial-columns.md:236]:
```suggestion
3. Alter the column type, as `serial` maps to `integer` (!= `bigint`).
```

File: doc/md/knowledge/postgres/serial-columns.md [hunk 16]
```diff
@@ -0,0 +1,258 @@
...
+
+We call [`atlas schema apply`](../../cli/reference.md#atlas-schema-apply) to plan and execute this three steps process
```
Comment (hilakashai) [inline doc/md/knowledge/postgres/serial-columns.md:238]:
```suggestion
We call [`atlas schema apply`](../../cli/reference.md#atlas-schema-apply) to plan and execute this three step process
```

File: doc/md/knowledge/postgres/serial-columns.md [hunk 17]
```diff
@@ -0,0 +1,258 @@
...
+ 3
+ 4
+// highlight-end
+```
+
+### Managing Serial Columns with Atlas
+
+Atlas makes it easier to define and manipulate columns of `serial` types. Let's use the
+[`atlas schema inspect`](../../cli/reference.md#atlas-schema-inspect) command to get a representation in Atlas 
+the Atlas language of the table we created above:
```
Comment (a8m) [inline doc/md/knowledge/postgres/serial-columns.md:120]:
```suggestion
[`atlas schema inspect`](../../cli/reference.md#atlas-schema-inspect) command to get a representation
of the table we created above in the Atlas HCL format :
```

**Pull request merged**